presto-main/src/main/java/com/facebook/presto/sql/planner/plan/IndexSourceNode.java

```diff
@@ -67,6 +67,11 @@ public IndexSourceNode(
         checkArgument(!outputSymbols.isEmpty(), "outputSymbols is empty");
         checkArgument(assignments.keySet().containsAll(lookupSymbols), "Assignments do not include all lookup symbols");
         checkArgument(outputSymbols.containsAll(lookupSymbols), "Lookup symbols need to be part of the output symbols");
+        Set<ColumnHandle> assignedColumnHandles = ImmutableSet.copyOf(assignments.values());
+        effectiveTupleDomain.getDomains().ifPresent(handleToDomain ->
+                checkArgument(
+                        assignedColumnHandles.containsAll(handleToDomain.keySet()),
+                        "Tuple domain handles must have assigned symbols"));
```

Comment:
can you extract the code you added as a method, so it will be easier to read this as you give a method name which would work a comment describing what it does